internal/git/provider/github_provider.go

```diff
@@ -1,16 +1,111 @@
 package provider
 
-import "golang.org/x/net/context"
+import (
+	"fmt"
 
-type GitHubProvider struct{}
+	"github.com/go-logr/logr"
+	"github.com/jenkins-x/go-scm/scm"
+	"github.com/jenkins-x/go-scm/scm/driver/github"
+	"github.com/jenkins-x/go-scm/scm/factory"
+	"golang.org/x/net/context"
+)
 
-func (p GitHubProvider) ListPullRequests(ctx context.Context, repo Repository) (_ []PullRequest) {
-	panic("not implemented") // TODO: Implement
+type GitHubProvider struct {
+	log      logr.Logger
+	apiToken string
+	hostname string
+	client   *scm.Client
 }
-func (p GitHubProvider) AddCommentToPullREquest(ctx context.Context, repo PullRequest, comment []byte) {
-	panic("not implemented") // TODO: Implement
+
+func (p GitHubProvider) ListPullRequests(ctx context.Context, repo Repository) ([]PullRequest, error) {
+	prList, _, err := p.client.PullRequests.List(ctx, repo.String(), &scm.PullRequestListOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("failed to list pull requests: %w", err)
+	}
+
+	prs := []PullRequest{}
+
+	for _, pr := range prList {
+		prs = append(prs, PullRequest{
+			Repository: repo,
+			Number:     pr.Number,
+			BaseBranch: pr.Base.Ref,
+			HeadBranch: pr.Head.Ref,
+			HeadRef:    pr.Head.Ref,
```

Comment:
Is this right? Both being assigned to the same value?

Reply:
Good catch

Reply:
Checked and one can be removed. I think originally I Ref in both cases, but renamed to Branch and left it there to rename it to Sha and populate with that value. So I did that instead of removing it.